app/src/main/java/com/rob729/quiethours/Fragments/MainFragment.kt

```diff
@@ -105,6 +109,36 @@ class MainFragment : Fragment() {
                     .navigate(MainFragmentDirections.actionMainFragmentToNewProfileFragment())
             }
         }
+        val sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+        val isVisible: Int = sharedPreferences.getInt("binding.activeProfile", View.GONE)
+        val setIcon: Int = sharedPreferences.getInt("binding.vibIcon", R.drawable.vibration)
+        val setActiveName: String? = sharedPreferences.getString("binding.activeName", "Active Profile: ")
+        val setEndTime: String? = sharedPreferences.getString("binding.end", "End Time: ")
+
+        binding.activeProfile.visibility = isVisible
+        binding.activeName.text = setActiveName
+        binding.end.text = setEndTime
+        binding.vibIcon.setImageResource(setIcon)
+
+        if (StartAlarm.begStatus == 1) {
+            binding.activeProfile.visibility = View.VISIBLE
+            sharedPreferences.edit().putInt("binding.activeProfile", View.VISIBLE).apply()
+            binding.activeName.text = "Active Profile: " + StartAlarm.activeProfile
+            sharedPreferences.edit().putString("binding.activeName", "Active Profile: " + StartAlarm.activeProfile).apply()
+            binding.end.text = "End Time: " + StartAlarm.etime + " hours"
+            sharedPreferences.edit().putString("binding.end", "End Time: " + StartAlarm.etime).apply()
```

Comment:
Define a const file in utils package and define all the string constants there used in the shared Preference.
Keep the naming convention like END_TIME_STRING instead of binding.end and similarly for other strings as well.